blackd.py

```diff
@@ -0,0 +1,93 @@
+import asyncio
+import logging
+
+from aiohttp import web
+import black
+import click
+
+# This is used internally by tests to shut down the server prematurely
+_stop_signal = asyncio.Event()
+
+VERSION_HEADER = "X-Protocol-Version"
+LINE_LENGTH_HEADER = "X-Line-Length"
+PYTHON_VARIANT_HEADER = "X-Python-Variant"
+SKIP_STRING_NORMALIZATION_HEADER = "X-Skip-String-Normalization"
+FAST_OR_SAFE_HEADER = "X-Fast-Or-Safe"
+
+
+@click.command(context_settings={"help_option_names": ["-h", "--help"]})
+@click.option(
+    "--bind-host", type=str, help="Address to bind the server to.", default="localhost"
+)
+@click.option("--bind-port", type=int, help="Port to listen on", default=45484)
+@click.version_option(version=black.__version__)
+def main(bind_host: str, bind_port: int) -> None:
+    logging.basicConfig(level=logging.INFO)
+    app = make_app()
+    ver = black.__version__
+    black.out(f"blackd version {ver} listening on {bind_host} port {bind_port}")
+    web.run_app(app, host=bind_host, port=bind_port, handle_signals=True, print=None)
+
+
+def make_app() -> web.Application:
+    app = web.Application()
+    app.add_routes([web.post("/", handle)])
+    return app
+
+
+async def handle(request: web.Request) -> web.Response:
+    try:
```

Comment:
This "try/except" block is too far reaching.

Reply:
Apparently any uncaught exceptions here will cause no response to be sent back to the client. Arguably this should be a middleware, and I'm happy to refactor it that way. Did you have any other concerns?